src/main/java/cceclipseplugin/editor/listeners/DirectoryListener.java

```diff
@@ -88,6 +89,16 @@ protected void handleFile(IResourceDelta delta) {
 		System.out.println( "	Filename: " + f.getName() + "	File flag: " + delta.getFlags());
 		
 		if (delta.getKind() == IResourceDelta.CHANGED) {
+			if (fileMeta == null) {
+				System.out.println("No metadata found for file change event, deleting file locally");
+				try {
+					f.delete(true, new NullProgressMonitor());
+				} catch (CoreException e) {
```

Comment:
Nope, the DirectoryListener would stop recursing down the resource change tree at the project node because